lib/simplecov/source_file.rb

```diff
@@ -82,8 +82,7 @@ def initialize(filename, coverage)
 
     # The path to this source file relative to the projects directory
     def project_filename
-      project_dir = File.dirname(File.expand_path(File.basename(__FILE__)))
-      @filename.sub(/^#{project_dir}/, "")
+      @filename.sub(/^#{SimpleCov.root}/, "")
```

Comment:
Agreed, but this change was kind of required for the regex. Without it the regex would be matching the full path, which probably nobody would understand.